lib/cbra/change.rb

```diff
@@ -9,14 +9,34 @@ def initialize(path, option, branch)
       @path = path
       @option = option
       @branch = branch
+      @tree = ComponentTree.new(path).to_h
     end
 
     def run!
       return unless valid?
+
+      changes_since_last_commit
+      directly_affected_components
     end
 
   private
 
+    def changes_since_last_commit
+      puts "<<<Changes since last commit on #{@branch}>>>"
+      puts blank_line
+      root_dir = `cd "#{@path}" && git rev-parse --show-toplevel`.chomp
+      @changes = `cd "#{root_dir}" && git diff --name-only #{@branch}`.split("\n").map { |f| File.join(root_dir, f) }
```

Comment:
Then do this:

```ruby
  def run!
    return unless valid?

    puts "<<<Changes since last commit on #{@branch}>>>"
    puts blank_line
    puts changes
  end

private

  def changes
    @changes ||= begin
      root_dir = `cd "#{@path}" && git rev-parse --show-toplevel`.chomp
      `cd "#{root_dir}" && git diff --name-only #{@branch}`.split("\n").map { |f| File.join(root_dir, f) }
    end
  end

  def directly_affected(parent_component)
    @directly_affected ||= []
    parent_component[:dependencies].each do |component|
      changes.each do |change|
        @directly_affected << component[:name] if change.start_with?(component[:path])
      end
      directly_affected(component)
    end
    @directly_affected.uniq!
  end
```